include/geometry/single_shape.h

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2023 UltiMaker
+// CuraEngine is released under the terms of the AGPLv3 or higher
+
+#ifndef GEOMETRY_SINGLE_SHAPE_H
+#define GEOMETRY_SINGLE_SHAPE_H
+
+#include "geometry/shape.h"
+
+namespace cura
+{
+
+class Polygon;
+
+/*!
+ * @brief A single area with holes. The first polygon is the outline, while the rest are holes within this outline.
+ * @sa https://github.com/Ultimaker/CuraEngine/wiki/Geometric-Base-Types#singleshape
+ *
+ * This class has little more functionality than Shape, but serves to show that a specific instance
+ * is ordered such that the first Polygon is the outline and the rest are holes.
+ */
+class SingleShape : public Shape
```

Comment:
I'm not entirely sure I see much use for this particular façade. Now that it's there I suggest to just leave it in. -- This comment was also brought to you by the feeling that there's a lot of fragmentation with all of these smaller classes; the things they represent get really specific!

Reply:
It existed, so I kept it :smile: but I'm no big fan of it either, especially because nothing ensures that what the documentation says is/remains true.
However, it adds a bit of explicity at some points, where you know just by reading the type what the object shall contain, so it does serve some purpose.